django_mongodb/operations.py

```diff
@@ -0,0 +1,123 @@
+            converters.append(self.convert_decimalfield_value)
+        elif internal_type == "TimeField":
+            converters.append(self.convert_timefield_value)
+        elif internal_type == "UUIDField":
+            converters.append(self.convert_uuidfield_value)
+        return converters
+
+    def convert_datefield_value(self, value, expression, connection):
+        if value is not None:
+            value = datetime.date.fromisoformat(value)
+        return value
+
+    def convert_datetimefield_value(self, value, expression, connection):
+        if value is not None:
+            value = datetime.datetime.fromisoformat(value)
+        return value
+
+    def convert_decimalfield_value(self, value, expression, connection):
+        if value is not None:
+            value = decimal.Decimal(value)
+        return value
+
+    def convert_timefield_value(self, value, expression, connection):
+        if value is not None:
+            value = datetime.time.fromisoformat(value)
+        return value
+
+    def convert_uuidfield_value(self, value, expression, connection):
+        if value is not None:
+            value = uuid.UUID(value)
+        return value
+
+    def prep_for_like_query(self, x):
+        # Override value escaping for LIKE queries.
+        return str(x)
+
+    def quote_name(self, name):
+        if name.startswith('"') and name.endswith('"'):
+            return name  # Quoting once is enough.
+        return name
+
+    def sql_flush(self, style, tables, *, reset_sequences=False, allow_cascade=False):
+        """
+        Return a list of the tables which will be passed as argument to
+        execute_sql_flush().
+        """
+        return tables
+
+    def execute_sql_flush(self, tables):
+        for table in tables:
+            if table.startswith("system."):
+                # Do not drop system collections.
+                continue
+
+            collection = self.connection.database[table]
+            options = collection.options()
+            if not options.get("capped", False):
+                collection.drop()
+
+    def prep_lookup_value(self, value, field, lookup):
+        """
+        Perform type-conversion on `value` before using as a filter parameter.
+
+        This precomputes `field`'s kind and a db_type for the field (or the
+        primary key of the related model for ForeignKeys etc.) and knows that
+        arguments to the `isnull` lookup (`True` or `False`) should not be
+        converted, while some other lookups take a list of arguments.
+        """
+        field, field_kind, db_type = self._convert_as(field, lookup)
+
+        # Argument to the "isnull" lookup is just a boolean, while some
+        # other lookups take a list of values.
+        if lookup == "isnull":
+            return value
+        if lookup in ("in", "range", "year"):
+            return [
+                self._prep_lookup_value(subvalue, field, field_kind, db_type, lookup)
```

Comment:
`DatabaseOperations.prep_lookup_value()` is called with `value` as a single value or a list of values. `_prep_lookup_value()` processes each value.

`prep_lookup_value()` isn't a standard Django method. It's what's left of Mongo Engine's [DatabaseOperations._value_from_db()](https://github.com/django-nonrel/mongodb-engine/blob/1314786975d3d17fb1a2e870dfba004158a5e431/django_mongodb_engine/base.py#L89). I started out copying the entire thing but later found much of it was unnecessary. It may be expanded or obsoleted as we continue to triage the Django test suite.